Sources/SmithyReadWrite/ReadingClosure.swift

```diff
@@ -5,4 +5,191 @@
+    return { reader in
+        try reader.readTimestamp(format: format)
+    }
+}
+
+public func timestampReadingClosure<Reader: SmithyReader>(format: TimestampFormat) -> ReadingClosure<Date?, Reader> {
+    return { reader in
+        try reader.readTimestampIfPresent(format: format)
+    }
+}
+
+public extension String {
+
+    static func read<Reader: SmithyReader>(from reader: Reader) throws -> String {
+        try reader.read()
+    }
+
+    static func read<Reader: SmithyReader>(from reader: Reader) throws -> String? {
+        try reader.readIfPresent()
+    }
+}
+
+public extension RawRepresentable where RawValue == Int {
+
+    static func read<Reader: SmithyReader>(from reader: Reader) throws -> Self {
+        try reader.read()
+    }
+
+    static func read<Reader: SmithyReader>(from reader: Reader) throws -> Self? {
+        try reader.readIfPresent()
+    }
+}
+
+public extension RawRepresentable where RawValue == String {
+
+    static func read<Reader: SmithyReader>(from reader: Reader) throws -> Self {
+        try reader.read()
+    }
+
+    static func read<Reader: SmithyReader>(from reader: Reader) throws -> Self? {
+        try reader.readIfPresent()
+    }
+}
+
+public extension Bool {
+
+    static func read<Reader: SmithyReader>(from reader: Reader) throws -> Bool {
+        try reader.read()
+    }
+
+    static func read<Reader: SmithyReader>(from reader: Reader) throws -> Bool? {
+        try reader.readIfPresent()
+    }
+}
+
+public extension Int {
+
+    static func read<Reader: SmithyReader>(from reader: Reader) throws -> Int {
+        try reader.read()
+    }
+
+    static func read<Reader: SmithyReader>(from reader: Reader) throws -> Int? {
+        try reader.readIfPresent()
+    }
+}
+
+public extension Float {
+
+    static func read<Reader: SmithyReader>(from reader: Reader) throws -> Float {
+        try reader.read()
+    }
+
+    static func read<Reader: SmithyReader>(from reader: Reader) throws -> Float? {
+        try reader.readIfPresent()
+    }
+}
+
+public extension Double {
+
+    static func read<Reader: SmithyReader>(from reader: Reader) throws -> Double {
+        try reader.read()
+    }
+
+    static func read<Reader: SmithyReader>(from reader: Reader) throws -> Double? {
+        try reader.readIfPresent()
+    }
+}
+
+public extension Data {
+
+    static func read<Reader: SmithyReader>(from reader: Reader) throws -> Data {
+        try reader.read()
+    }
+
+    static func read<Reader: SmithyReader>(from reader: Reader) throws -> Data? {
+        try reader.readIfPresent()
+    }
+}
+
+public extension Document {
+
+    static func read<Reader: SmithyReader>(from reader: Reader) throws -> Document {
+        try reader.read()
+    }
+
+    static func read<Reader: SmithyReader>(from reader: Reader) throws -> Document? {
+        try reader.readIfPresent()
+    }
+}
+
+public func optionalFormOf<T, Reader: SmithyReader>(
```

Comment:
This function is used to turn a reading closure from non-optional to optional.  Used with sparse collections to correctly read null elements.